packages/dd-trace/src/native/metrics/main.cc

```diff
@@ -0,0 +1,477 @@
+// windows.h defines min and max macros.
```

Comment:
Why put everything in a single file? It makes it more difficult to find, review and organize code.

Reply:
For smaller classes I tend to avoid headers just because of the redundancy. The file is under 500 lines which seems pretty reasonable to me but I can split it up if you want.

Reply:
One thing to consider is that it took quite a bit of time to compile all the files (not too much on Linux, maybe 10 seconds, but on Windows it was about a minute). If having a single file would reduce these times then I think it's worth it, but otherwise I definitely prefer multiple files. At the same time I'm not the best reference for C++ so I don't know how common single files are.

@bengl Thoughts?

Reply:
I think it's fine in one file. This is small enough that having it all in one editor window is pretty useful too. I also am 💯 on the point about the headers, especially when these things are so (relatively) tiny.